test/topology_custom/test_mv_delete_partitions.py

```diff
@@ -0,0 +1,82 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+from test.pylib.manager_client import ManagerClient
+
+import asyncio
+import pytest
+import time
+import logging
+from test.topology.conftest import skip_mode
+from cassandra.cqltypes import Int32Type
+
+logger = logging.getLogger(__name__)
+
+async def insert_with_concurrency(cql, values, concurrency):
+    def serialize_int(i):
+        return Int32Type.serialize(i, cql.cluster.protocol_version)
+    def get_replicas(key):
+        return cql.cluster.metadata.get_replicas("ks", serialize_int(key))
+    local_node = get_replicas(0)[0]
+    logger.info(f"Starting writes with concurrency {concurrency}")
+    async def do_inserts(i: int):
+        # Insert all rows with keys with the same modulo as i
+        insert_stmt = cql.prepare(f"INSERT INTO ks.tab (key, c, v) VALUES (?, ?, ?)")
+        i_steps = i
+        while i_steps < values:
+            # Only remote updates hold on to memory, so make all updates remote
+            while local_node == get_replicas(i)[0]:
+                i = i + concurrency
+            await cql.run_async(insert_stmt, [0, i, i])
+            i_steps += concurrency
+            i = i + concurrency
+    tasks = [asyncio.create_task(do_inserts(i)) for i in range(concurrency)]
+
+    await asyncio.gather(*tasks)
+    logger.info(f"Finished writes with concurrency {concurrency}")
+
+async def wait_for_views(cql, mvs_count, node_count):
+    deadline = time.time() + 120
+    while time.time() < deadline:
+        done = await cql.run_async(f"SELECT COUNT(*) FROM system_distributed.view_build_status WHERE status = 'SUCCESS' ALLOW FILTERING")
+        logger.info(f"Views built: {done[0][0]}")
+        if done[0][0] == node_count * mvs_count:
+            return
+        else:
+            time.sleep(0.2)
+    raise Exception("Timeout waiting for views to build")
+
+# This test reproduces issue #12379
+# To quickly exceed the view update backlog limit, the test uses a minimal, 2 node
+# cluster and lowers the limit using the "view_update_limit" error injection.
+# Also, the "delay_before_remote_view_update" error injection is used to ensure that
+# we're hitting a scenario where the remote view update finishes after the base write
+# returns.
+# The test steps are:
+# 1. Create a table and with one partition and 200 rows
```

Comment:
So, I have to say I don't understand... You only have one partition and 200 rows, so at worst even if you do all view updates together, there are 200 view updates pending, each a small deletion. How could these ever finish the 250,000 bytes you set for the view update backlog limit?

Reply:
I'm taking into account the overhead of view updates here, which is slightly over 2KB, so with 200 view updates we exceed the limit on the second batch.

It's hard to make it independent of the overhead because of the batch size - if we make the row so large that the overhead is a minor fraction then in the first batch, #17777 will kick in and the replica write will fail due to that. We also don't want to increase the limit so that a batch of larger updates will fit because the test will take a lot longer because of that.
A possible solution is changing the view update batch size using e.g. an injection, but maybe the current version is good enough.

Reply:
Oh, I finally understand, thanks. So the numbers 200 and 250,000 are important for three reasons:
1. 200 > 100 so we'll get two batches and the possibility of a wait between them.
2. 100*2000 < 250000 so the first batch fits (each request gets 2000 bytes overhead), but
3. If we don't wait enough between batches and do the second one immediately, 200*2000 > 250000 and the test will fail (the injection makes the overflow into a failure).

What worries me now is that this test may not work if running very slow... The sleep between the first and second batch is constant time, but the progress made during that sleep depends on the speed if the machine (i think). If not enough progress is made during that sleep, the second batch will fail (because of the injection that asks it to fail on overflow).

Reply:
> Oh, I finally understand, thanks. So the numbers 200 and 250,000 are important for three reasons:
> 
> 1. 200 > 100 so we'll get two batches and the possibility of a wait between them.
> 2. 100*2000 < 250000 so the first batch fits (each request gets 2000 bytes overhead), but
> 3. If we don't wait enough between batches and do the second one immediately, 200*2000 > 250000 and the test will fail (the injection makes the overflow into a failure).

Everything sounds correct, what's worth adding is that 100*2000 (actually closer to 100*2400) is also near 250000 so the throttle delay is relatively long.

> What worries me now is that this test may not work if running very slow... The sleep between the first and second batch is constant time, but the progress made during that sleep depends on the speed if the machine (i think). If not enough progress is made during that sleep, the second batch will fail (because of the injection that asks it to fail on overflow).

That's a possibility, though I'm not sure how realistic, we have 100 small writes to do in ~0.4s (~0.9s due to backlog fullness minus 500ms from `delay_before_remote_view_update` injection)
If we're still concerned about it we can also decrease the number of rows in the second batch. I used 200 there but we really want any number that will exceed the limit together with the second batch. If we do that we won't have to finish 100 writes, just as many as were exceeding the limit